tests/Get-DbaPermission.Tests.ps1

```diff
@@ -4,38 +4,141 @@ Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
-            $results.where( {$_.Database -eq ''}).count | Should BeGreaterThan 0
+        It "returns server level permissions with -IncludeServerLevel" {
+            $results = Get-DbaPermission -SqlInstance $server -IncludeServerLevel
+            $results.where( { $_.Database -eq '' }).count | Should BeGreaterThan 0
+        }
+        It "returns no server level permissions without -IncludeServerLevel" {
+            $results = Get-DbaPermission -SqlInstance $server
+            $results.where( { $_.Database -eq '' }).count | Should Be 0
         }
-        it "returns no server level permissions without -IncludeServerLevel" {
-            $results = Get-DbaPermission -SqlInstance $script:instance2
-            $results.where( {$_.Database -eq ''}).count | Should Be 0
+        It "returns no system object permissions with -ExcludeSystemObjects" {
+            $results = Get-DbaPermission -SqlInstance $server -ExcludeSystemObjects
+            $results.where( { $_.securable -like 'sys.*' }).count | Should Be 0
         }
-        it "returns no system object permissions with -ExcludeSystemObjects" {
-            $results = Get-DbaPermission -SqlInstance $script:instance2 -ExcludeSystemObjects
-            $results.where( {$_.securable -like 'sys.*'}).count | Should Be 0
+        It "returns system object permissions without -ExcludeSystemObjects" {
+            $results = Get-DbaPermission -SqlInstance $server
+            $results.where( { $_.securable -like 'sys.*' }).count | Should BeGreaterThan 0
         }
-        it "returns system object permissions without -ExcludeSystemObjects" {
-            $results = Get-DbaPermission -SqlInstance $script:instance2
-            $results.where( {$_.securable -like 'sys.*'}).count | Should BeGreaterThan 0
+        It "db object level permissions for a user are returned correctly" {
+            $results = Get-DbaPermission -SqlInstance $server -Database $dbName -ExcludeSystemObjects | Where-Object { $_.Grantee -eq $loginNameUser1 -and $_.SecurableType -ne "SCHEMA" }
+            $results.count | Should -Be 4
+            $results.where( { $_.Securable -eq "dbo.$tableName1" -and $_.PermState -eq 'DENY' -and $_.PermissionName -in ('DELETE', 'INSERT', 'UPDATE') }).count | Should -Be 3
+            $results.where( { $_.Securable -eq "dbo.$tableName1" -and $_.PermState -eq 'GRANT' -and $_.PermissionName -eq 'SELECT' }).count | Should -Be 1
         }
     }
     Context "Validate input" {
-        it "Cannot resolve hostname of computer" {
-            mock Resolve-DbaNetworkName {$null}
-            {Get-DbaComputerSystem -ComputerName 'DoesNotExist142' -WarningAction Stop 3> $null} | Should Throw
+        It "Cannot resolve hostname of computer" {
+            Mock Resolve-DbaNetworkName { $null }
+            { Get-DbaComputerSystem -ComputerName 'DoesNotExist142' -WarningAction Stop 3> $null } | Should Throw
+        }
+    }
+
+    Context "Bug 6744" {
```

Comment:
We need a statement here instead of just putting the issue number. A comment can be added with the link directly to the issue.

Reply:
ok, will do